mne/epochs.py

```diff
@@ -388,12 +394,13 @@ def decimate(self, decim, copy=False):
                  'of %g Hz. The decim=%i parameter will result in a sampling '
                  'frequency of %g Hz, which can cause aliasing artifacts.'
                  % (lowpass, decim, new_sfreq))  # > 50% nyquist lim
-
+        offset = int(offset) % decim
         epochs._decim *= decim
         start_idx = int(round(epochs._raw_times[0] * (epochs.info['sfreq'] *
                                                       epochs._decim)))
         i_start = start_idx % epochs._decim
-        decim_slice = slice(i_start, len(epochs._raw_times), epochs._decim)
+        decim_slice = slice(i_start + offset, len(epochs._raw_times),
```

Comment:
no, you get at most `epochs._raw_times` out (indices should never go < 0 or >= len(epochs._raw_times))